### ISSUE TITLE
Events

### ISSUE THREAD

**Issue by BEZEI2K** [orchestrate-io/gorc2#4]
I might of missed it, but fetching an event based on being between to timestamps...

**Comment by liquidgecka** [comment]
Man I need to add a bunch more examples. I keep meaning to get that done and end up getting distracted on other things.. So here is an example of how to do this:

```go
client := gorc2.NewClient("API_KEY")
collection := client.Collection("COLLECTION")
iterator := collection.ListEvents("key", "type", &gorc2.ListEventsQuery{
    PageSize: 100,
    Start: time.Unix(startSec, startNsec),
    End: time.Unix(endSec, endNsec),
})
for iterator.Next() {
    var value interface{}
    event, err := iterator.GetEvent(&value)
    if err != nil {
        // Do something here
    }
}
if iterator.Error != nil {
    // Do something here
}
```

A couple of notes on this, though time.Unix takes nano seconds the Orchestrate api only takes milliseconds, as such any precision beyond the millisecond level is lost. (time.Unix(0, 1) and time.Unix(0, 2) are the same because they both fall in the same millisecond).

Also the search is [start, end) meaning that events in the ending timestamp will not be included. If you wish to include them you can set EndOrdinal to math.MaxInt64 which makes the query [start, end]. This is due to the way that ordinals are processed.

Basically internally an event looks like this:
[time stamp, ordinal], [timestamp, ordinal]...

So querying from [start -> end] is actually searching from [(start, 0) -> (end, 0)] so any event with end as a timestamp would actually be (end, ordinal) where ordinal is greater than zero, and therefor it will NOT be included in the range. Hence the trick above, making the search from [(start, 0) -> (end, MaxInt64)].

Hopefully that is not just making it far more confusing. Let me know if any of this doesn't make sense. =)

[For now I am going to consider this issue as "provide an example of this you lazy sod" and close it when I finally get around to committing the example.]

**Comment by BEZEI2K** [comment]
Awesome, I actually dug into, and figured out the ListEvents was what I was looking for. Its ok, my CTO wrote a perl orchestrate module that we have yet to document. lol. So I feel the pain.
